samples/scala-valueentity-counter/src/test/scala/com/example/domain/CounterSpec.scala

```diff
@@ -0,0 +1,57 @@
+/* This code was generated by Akka Serverless tooling.
+ * As long as this file exists it will not be re-generated.
+ * You are free to make changes to this file.
+ */
+package com.example.domain
+
+import com.example.{DecreaseValue, IncreaseValue, ResetValue}
+import com.google.protobuf.empty.Empty
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+// tag::sample-unit-test[]
+class CounterSpec
+    extends AnyWordSpec
+    with Matchers {
+
+  "Counter" must {
+
+    "handle command Increase" in {
+      val testKit = CounterTestKit(new Counter(_))
+
+      val result1 = testKit.increase(IncreaseValue(value = 1))
+      result1.getReply should === (Empty.defaultInstance)
```

Comment:
We should decide what assert matchers to use. In Akka we used `===` for a while because we wanted to have the compile time type check, but I'm not sure that is working here (without adding more stuff)?

I'd suggest we use the more simplistic
```
result1.getReply should shouldBe Empty.defaultInstance
```

Reply:
The test passes with any further imports etc. but I agree, the tripple equals is anyway a bit funky.

One thing I thought about: should we make the testkit more Scala:y with a `.reply` instead of `.getReply` and `.state` instead of `.getState` when we anyway have specific types?

Reply:
> The test passes with any further imports etc. but I agree, the tripple equals is anyway a bit funky.

I mean, does it find type errors at compile time? This shouldn't compile if it should have any added value
```
result1.getReply should === ("foo")
```

One thing I thought about: should we make the testkit more Scala:y

That would be better for the Result, but for the generated testkit we have `getState()` and that might be somewhat confusing as `state` or `state()` since it's changing. We could name that `currentState()` if we prefer that over `getState()`?

Reply:
I do like `currentState()` better.

Reply:
Changed, `getReply()` => `reply`, `getState()` => `currentState()`